module/src/main/scala/com/gu/googleauth/auth.scala

```diff
@@ -108,6 +108,10 @@ object GoogleAuth {
           googleResponse(response) { json =>
               val token = Token.fromJson(json)
               val jwt = token.jwt
+              config.domain foreach { domain =>
+                if (!jwt.claims.email.split("@").lastOption.contains(domain))
```

Comment:
:+1:  for an exact match.

Also UserIdentity is already extracting the domain, see https://github.com/guardian/play-googleauth/blob/d4b6085203c75ce8576cce91d1cfc8a4453bc92f/module/src/main/scala/com/gu/googleauth/actions.scala#L11, so in the mid-term, I'd suggest we use that, but I'm happy to merge this now and tidy later.